core/modules/qmeta/QMetaMysql.cc

```diff
@@ -0,0 +1,569 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "qmeta/QMetaMysql.h"
+
+// System headers
+#include <algorithm>
+
+// Third-party headers
+#include "boost/lexical_cast.hpp"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "Exceptions.h"
+#include "QMetaTransaction.h"
+#include "sql/SqlResults.h"
+
+namespace {
+
+// logger instance for this module
+LOG_LOGGER _logger = LOG_GET("lsst.qserv.qmeta.QMetaMysql");
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace qmeta {
+
+// Constructors
+QMetaMysql::QMetaMysql(mysql::MySqlConfig const& mysqlConf)
+  : QMeta(),
+    _conn(mysqlConf)
+{
+    // Check that database is in consistent state
+    _checkDb();
+}
+
+// Destructor
+QMetaMysql::~QMetaMysql()
+{
+}
+
+/**
+ *  @brief Return czar ID given czar "name".
+ *
+ *  Negative number is returned if czar does not exist.
+ *
+ *  @param name:  Czar name, arbitrary string.
+ *  @return: Car ID, negative if czar does not exist.
+ */
+int
+QMetaMysql::getCzarID(std::string const& name)
+{
+    QMetaTransaction trans(_conn);
+
+    // run query
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    const std::string query = "SELECT cid from QCzar where czar = '" + name +"'";
+    LOGF(_logger, LOG_LVL_DEBUG, "Executing query: %1%" % query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "SQL query failed: %1%" % query);
+        throw SqlError(ERR_LOC, errObj);
+    }
+
+    // get results of the query
+    std::vector<std::string> ids;
+    if (not results.extractFirstColumn(ids, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "Failed to extract czar ID from query result");
+        throw SqlError(ERR_LOC, errObj);
+    }
+
+    trans.commit();
+
+    // check number of results and convert to integer
+    if (ids.empty()) {
+        return -1;
+    } else if (ids.size() > 1) {
+        throw ConsistencyError(ERR_LOC, "More than one czar ID found for czar name " + name +
+                               ": " + boost::lexical_cast<std::string>(ids.size()));
+    } else {
+        return boost::lexical_cast<int>(ids[0]);
+    }
+}
+
+// Register new czar, return czar ID.
+int
+QMetaMysql::registerCzar(std::string const& name)
+{
+    QMetaTransaction trans(_conn);
+
+    // check if czar is already defined
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    std::string query = "SELECT cid from QCzar where czar = '" + name +"'";
+    LOGF(_logger, LOG_LVL_DEBUG, "Executing query: %1%" % query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "SQL query failed: %1%" % query);
+        throw SqlError(ERR_LOC, errObj);
```

Comment:
ps I saw that you are aborting in destructor, so I guess we don't need to call abort explicitly here.